src/hotspot/share/opto/superword.cpp

```diff
@@ -2359,472 +2052,6 @@ void SuperWord::schedule_reorder_memops(Node_List &memops_schedule) {
-      // Vector instructions do not mask shift count, do it here.
-      juint mask = (p0->bottom_type() == TypeInt::INT) ? (BitsPerInt - 1) : (BitsPerLong - 1);
-      const TypeInt* t = opd->find_int_type();
-      if (t != nullptr && t->is_con()) {
-        juint shift = t->get_con();
-        if (shift > mask) { // Unsigned cmp
-          cnt = igvn().intcon(shift & mask);
-          phase()->set_ctrl(cnt, phase()->C->root());
-        }
-      } else {
-        if (t == nullptr || t->_lo < 0 || t->_hi > (int)mask) {
-          cnt = igvn().intcon(mask);
-          cnt = new AndINode(opd, cnt);
-          phase()->register_new_node_with_ctrl_of(cnt, opd);
-        }
-        if (!opd->bottom_type()->isa_int()) {
-          assert(false, "int type only");
-          return nullptr;
-        }
-      }
-      // Move shift count into vector register.
-      cnt = VectorNode::shift_count(p0->Opcode(), cnt, vlen, velt_basic_type(p0));
-      phase()->register_new_node_with_ctrl_of(cnt, opd);
-      return cnt;
-    }
-    if (opd->is_StoreVector()) {
-      assert(false, "StoreVector is not expected here");
-      return nullptr;
-    }
-    // Convert scalar input to vector with the same number of elements as
-    // p0's vector. Use p0's type because size of operand's container in
-    // vector should match p0's size regardless operand's size.
-    const Type* p0_t = nullptr;
-    VectorNode* vn = nullptr;
-    if (opd_idx == 2 && VectorNode::is_scalar_rotate(p0)) {
-       Node* conv = opd;
-       p0_t =  TypeInt::INT;
-       if (p0->bottom_type()->isa_long()) {
-         p0_t = TypeLong::LONG;
-         conv = new ConvI2LNode(opd);
-         phase()->register_new_node_with_ctrl_of(conv, opd);
-       }
-       vn = VectorNode::scalar2vector(conv, vlen, p0_t);
-    } else {
-       p0_t =  velt_type(p0);
-       vn = VectorNode::scalar2vector(opd, vlen, p0_t);
-    }
-
-    phase()->register_new_node_with_ctrl_of(vn, opd);
-    VectorNode::trace_new_vector(vn, "SuperWord");
-    return vn;
-  }
-
-  // Insert pack operation
-  BasicType bt = velt_basic_type(p0);
-  PackNode* pk = PackNode::make(opd, vlen, bt);
-  DEBUG_ONLY( const BasicType opd_bt = opd->bottom_type()->basic_type(); )
-
-  for (uint i = 1; i < vlen; i++) {
-    Node* pi = p->at(i);
-    Node* in = pi->in(opd_idx);
-    if (get_pack(in) != nullptr) {
-      assert(false, "Should already have been unpacked");
-      return nullptr;
-    }
-    assert(opd_bt == in->bottom_type()->basic_type(), "all same type");
-    pk->add_opd(in);
-    if (VectorNode::is_muladds2i(pi)) {
-      Node* in2 = pi->in(opd_idx + 2);
-      if (get_pack(in2) != nullptr) {
-        assert(false, "Should already have been unpacked");
-        return nullptr;
-      }
-      assert(opd_bt == in2->bottom_type()->basic_type(), "all same type");
-      pk->add_opd(in2);
-    }
-  }
-  phase()->register_new_node_with_ctrl_of(pk, opd);
-  VectorNode::trace_new_vector(pk, "SuperWord");
-  return pk;
```

Comment:
Note: this `PackNode` code seems to be dead code, if you read `SuperWord::profitable` then there are comments saying that we are not using `PackNode`. I replaced it with a `ShouldNotReachHere` in `SuperWordVTransformBuilder::find_input_for_vector`.